contracts/Swaplace.sol

```diff
@@ -36,7 +36,9 @@ contract Swaplace is SwapFactory, ISwaplace, IERC165 {
 
     _swaps[swapId] = swap;
 
-    emit SwapCreated(swapId);
+    (address allowed,) = parseData(swap.config);
+
+    emit SwapCreated(swapId, swap.owner, allowed);
```

Comment:
We use `msg.sender` instead of the swap.owner, because the gas consumption will be lower.